website/docs/r/mobile_network_attached_data_network.html.markdown

```diff
@@ -0,0 +1,200 @@
+  location            = azurerm_resource_group.example.location
+  mobile_country_code = "001"
+  mobile_network_code = "01"
+}
+
+resource "azurerm_mobile_network_packet_core_control_plane" "example" {
+  name                              = "example-mnpccp"
+  resource_group_name               = azurerm_resource_group.example.name
+  location                          = "West Europe"
+  sku                               = "G0"
+  mobile_network_id                 = azurerm_mobile_network.example.id
+  control_plane_access_name         = "default-interface"
+  control_plane_access_ipv4_address = "192.168.1.199"
+  control_plane_access_ipv4_gateway = "192.168.1.1"
+  control_plane_access_ipv4_subnet  = "192.168.1.0/25"
+
+  platform {
+    type           = "AKS-HCI"
+    edge_device_id = azurerm_databox_edge_device.example.id
+  }
+}
+
+resource "azurerm_mobile_network_packet_core_data_plane" "example" {
+  name                                        = "example-mnpcdp"
+  mobile_network_packet_core_control_plane_id = azurerm_mobile_network_packet_core_control_plane.example.id
+  location                                    = azurerm_resource_group.example.location
+  user_plane_access_name                      = "default-interface"
+  user_plane_access_ipv4_address              = "192.168.1.199"
+  user_plane_access_ipv4_gateway              = "192.168.1.1"
+  user_plane_access_ipv4_subnet               = "192.168.1.0/25"
+}
+
+resource "azurerm_mobile_network_data_network" "example" {
+  name              = "example-data-network"
+  mobile_network_id = azurerm_mobile_network.example.id
+  location          = azurerm_resource_group.example.location
+}
+
+resource "azurerm_mobile_network_attached_data_network" "example" {
+  name                                        = "example-data-network"
+  mobile_network_packet_core_data_plane_id    = azurerm_mobile_network_packet_core_data_plane.example.id
+  location                                    = azurerm_resource_group.example.location
+  dns_addresses                               = ["1.1.1.1"]
+  user_equipment_address_pool_prefixes        = ["2.4.1.0/24"]
+  user_equipment_static_address_pool_prefixes = ["2.4.2.0/24"]
+  user_plane_access_name                      = "test"
+  user_plane_access_ipv4_address              = "10.204.141.4"
+  user_plane_access_ipv4_gateway              = "10.204.141.1"
+  user_plane_access_ipv4_subnet               = "10.204.141.0/24"
+
+  network_address_port_translation_configuration {
+    enabled                = true
+    pinhole_maximum_number = 65536
+    pinhole_timeouts_in_seconds {
+      icmp = 30
+      tcp  = 100
+      udp  = 39
+    }
+    port_range {
+      max_port = 49999
+      min_port = 1024
+    }
+    port_reuse_minimum_hold_time_in_seconds {
+      tcp = 120
+      udp = 60
+    }
+  }
+
+  tags = {
+    key = "value"
+  }
+
+  depends_on = [azurerm_mobile_network_data_network.example]
```

Comment:
We can just specify the `name = azurerm_mobile_network_data_network.example.name` to introduce this dependency, as the `name` of this resource *Must be as same as `azurerm_mobile_network_data_network`*

Reply:
again, this was marked as resolved with no explanation, can we have context here?